File: src/main/java/spring/springserver/domain/chat/service/ChatService.java
package spring.springserver.domain.chat.service;

import spring.springserver.domain.chat.data.request.CreateChatRoomRequest;
import spring.springserver.domain.chat.data.request.SendChatMessageRequest;
import spring.springserver.domain.chat.data.response.ChatMessageResponse;
import spring.springserver.domain.chat.data.response.ChatRoomResponse;
import spring.springserver.domain.chat.data.response.CreateChatRoomResponse;

import java.util.List;

public interface ChatService {

    CreateChatRoomResponse createOrGetDirectRoom(String requesterUsername,
                                                 CreateChatRoomRequest request);

    List<ChatRoomResponse> getMyChatRooms(String username);

    List<ChatMessageResponse> getRoomMessages(String username,
                                              Long roomId);

    ChatMessageResponse sendMessage(String senderUsername,
                                    SendChatMessageRequest request);

    boolean canAccessRoom(String username,
                          Long roomId);
}

File: src/main/java/spring/springserver/domain/chat/service/ChatService.kt
package spring.springserver.domain.chat.service

import spring.springserver.domain.chat.data.request.CreateChatRoomRequest
import spring.springserver.domain.chat.data.request.SendChatMessageRequest
import spring.springserver.domain.chat.data.response.ChatMessageResponse
import spring.springserver.domain.chat.data.response.ChatParticipantResponse
import spring.springserver.domain.chat.data.response.ChatPaymentResponse
import spring.springserver.domain.chat.data.response.ChatRoomResponse
import spring.springserver.domain.chat.data.response.CreateChatRoomResponse

interface ChatService {

    fun createOrGetDirectRoom(
        requesterUsername: String,
        createChatRoomRequest: CreateChatRoomRequest
    ): CreateChatRoomResponse

    fun getMyChatRooms(
        username: String
    ): List<ChatRoomResponse>

    fun getRoomMessages(
        username: String,
        roomId: Long
    ): List<ChatMessageResponse>

    fun sendMessage(
        senderUsername: String,
        sendChatMessageRequest: SendChatMessageRequest
    ): ChatMessageResponse

    fun getOtherParticipantUsername(
        roomId: Long,
        username: String
    ): ChatParticipantResponse

    fun canAccessRoom(
        username: String,
        roomId: Long
    ): Boolean

    fun leaveRoom(
        username: String,
        roomId: Long
    )

    fun findDirectRoomId(
        clientId: Long,
        professionalId: Long,
        postId: Long
    ): Long?

    fun isRoomParticipant(
        roomId: Long,
        memberId: Long
    ): Boolean

    fun sendPaymentMessage(
        roomId: Long,
        senderMemberId: Long,
        payment: ChatPaymentResponse
    ): ChatMessageResponse
}
